core/http/src/tls.rs

```diff
@@ -125,4 +151,25 @@ impl Connection for TlsStream<TcpStream> {
     fn remote_addr(&self) -> Option<SocketAddr> {
         self.get_ref().0.remote_addr()
     }
+
+    fn peer_certificates(&self) -> Option<Vec<Certificate>> {
+        (self.get_ref().1).get_peer_certificates()
+    }
+}
+
+/// Mutual TLS information.
+/// This struct contains client certificate itself and
+/// its trust chain. Additionally, several useful certificate fields
+/// are exposed for your convenience.
+#[derive(Debug)]
+pub struct MutualTlsInfo {
```

Comment:
This is a lot of public fields; I would be happier exposing functions that return `&[u8]` or `Vec<u8>`.